File: Programmers/Lv0/가까운수.py
def solution(array, n):
    array.append(n)
    array.sort()
    if (max(array) == n):
        return array[len(array) - 2]
    elif (min(array) == n):
        return array[1]
    else:
        idx = array.index(n)
        if array[idx + 1] - array[idx] >= array[idx] - array[idx - 1]:
            return array[idx - 1]
        else:
            return array[idx + 1]

File: Programmers/Lv0/가까운수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> array, int n) {
    array.push_back(n);
    sort(array.begin(), array.end());
    auto it_max = max_element(array.begin(), array.end());
    auto it_min = min_element(array.begin(), array.end());

    if (*it_max == n)
        return array[array.size() - 2];
    else if (*it_min == n)
        return array[1];
    else {
        auto it_mid = find(array.begin(), array.end(), n);
        if (*(it_mid + 1) - *it_mid >= *it_mid - *(it_mid - 1))
            return *(it_mid - 1);
        else
            return *(it_mid + 1);
    }
}